src/app/share/views/SharedLinksView/components/SharedItemList.tsx

```diff
@@ -0,0 +1,243 @@
+import { useCallback } from 'react';
+import List from '../../../../shared/components/List';
+import { AdvancedSharedItem } from '../../../types';
+import { DriveFileData } from '../../../../drive/types';
+import iconService from '../../../../drive/services/icon.service';
+import usersIcon from 'assets/icons/users.svg';
+import Avatar from '../../../../shared/components/Avatar';
+import sizeService from '../../../../drive/services/size.service';
+import dateService from '../../../../core/services/date.service';
+import { t } from 'i18next';
+import { ListItemMenu } from '../../../../shared/components/List/ListItem';
+
+const skinSkeleton = [
+  <div key="1" className="flex flex-row items-center space-x-4">
+    <div key="2" className="h-8 w-8 rounded-md bg-gray-5" />
+    <div key="3" className="h-4 w-40 rounded bg-gray-5" />
+  </div>,
+  <div key="4" className="h-4 w-20 rounded bg-gray-5" />,
+  <div key="5" className="h-4 w-24 rounded bg-gray-5" />,
+  <div key="6" className="h-4 w-20 rounded bg-gray-5" />,
+];
+
+type SharedListItem = {
+  onClickItem: (item) => void;
+  keyBoardShortcutActions: {
+    onShiftFKeysPressed?: () => void;
+    onRKeyPressed?: () => void;
+    onBackspaceKeyPressed?: () => void;
+  };
+  emptyStateElement: JSX.Element;
+  shareItems: AdvancedSharedItem[];
+  isLoading: boolean;
+  disableKeyboardShortcuts;
+  selectedItems: AdvancedSharedItem[];
+  onSelectedItemsChanged: (changes: { props: AdvancedSharedItem; value: boolean }[]) => void;
+  onItemDoubleClicked: (shareItem: AdvancedSharedItem) => void;
+  onNameClicked: (shareItem: AdvancedSharedItem) => void;
+  onNextPage: () => void;
+  hasMoreItems: boolean;
+  contextMenu: ListItemMenu<AdvancedSharedItem> | undefined;
+  currentShareOwnerAvatar: string;
+  user?: AdvancedSharedItem['user'];
+};
+
+export const SharedItemList = ({
+  onClickItem,
+  keyBoardShortcutActions,
+  emptyStateElement,
+  shareItems,
+  isLoading,
+  disableKeyboardShortcuts,
+  selectedItems,
+  onSelectedItemsChanged,
+  onItemDoubleClicked,
+  onNameClicked,
+  onNextPage,
+  hasMoreItems,
+  contextMenu,
+  currentShareOwnerAvatar,
+  user,
+}: SharedListItem) => {
+  const getOwnerAvatarSrc = useCallback(
+    (shareItem: AdvancedSharedItem): string | null => {
+      if (currentShareOwnerAvatar) {
+        return currentShareOwnerAvatar;
+      }
+      return shareItem.user?.avatar ? shareItem.user?.avatar : null;
+    },
+    [currentShareOwnerAvatar],
+  );
+
+  const checkIfIsItemSelected = (item: AdvancedSharedItem) => {
+    return selectedItems.some((i) => item.id === i.id);
+  };
+
+  const itemComposition = getSharedListItemComposition({
+    onItemDoubleClicked,
+    onNameClicked,
+    user,
+    getOwnerAvatarSrc,
+    checkIfIsItemSelected,
+  });
+
+  return (
+    <List<any, 'updatedAt' | 'createdAt' | 'ownerId' | 'fileSize'>
```

Comment:
why any?

Reply:
It was already like that and I guess I forgot to change it. I'll check it!